crates/chia-sdk-driver/src/primitives/cat.rs

```diff
@@ -462,6 +459,55 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn test_different_cat_p2_puzzles() -> anyhow::Result<()> {
+        let mut sim = Simulator::new();
+        let ctx = &mut SpendContext::new();
+        let (sk, pk, puzzle_hash, coin) = sim.new_p2(2)?;
+
+        // This will just return the solution verbatim.
+        let custom_p2 = ctx.alloc(&1)?;
```

Comment:
you can also get multiple sks via `test_secret_keys(2)`

Reply:
But I only need one